pages/admin/reports/components/DateInputs.tsx

```diff
@@ -0,0 +1,219 @@
+import { useTreatments } from '@/hooks/useTreatments';
+import { StatisticsData } from '@/interfaces/statistics/StatisticsData';
+import { StatisticsDataResponse } from '@/interfaces/statistics/StatisticsDataResponse';
+import { getStatistics } from '@/services/StatisticsService';
+import {
+    Text,
+    Box,
+    Flex,
+    Input,
+    SelectRoot,
+    SelectTrigger,
+    SelectHiddenSelect,
+    SelectControl,
+    SelectIndicator,
+    SelectIndicatorGroup,
+    Portal,
+    SelectPositioner,
+    SelectValueText,
+    SelectItemIndicator,
+} from '@chakra-ui/react';
+
+import { createListCollection } from '@chakra-ui/react';
+import { SelectLabel, SelectContent, SelectItem } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+
+interface DateInputsProps {
+    onDateChange: (dates: StatisticsData) => void;
+    onStatisticsFetched: (data: StatisticsDataResponse) => void;
+    onTreatmentSelect: (treatmentId: string) => void;
+    onLoadingChange?: (isLoading: boolean) => void;
+}
+
+export default function DateInputs({
+    onStatisticsFetched,
+    onDateChange,
+    onTreatmentSelect,
+    onLoadingChange,
+}: DateInputsProps) {
+    const { treatments, error } = useTreatments();
+    const [formData, setFormData] = useState<StatisticsData>({
+        startDate: '',
+        endDate: '',
+        treatment: '',
+    });
+    const [selectTreatment, setSelectTreatment] = useState('');
+    const [statisticsLoading, setStatisticsLoading] = useState(false);
+    const [statisticsError, setStatisticsError] = useState<string | null>(null);
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = e.target;
+        setFormData((prev) => ({
+            ...prev,
+            [name]: value,
+        }));
+    };
+
+    const handleSelectChange = (value: string) => {
+        setSelectTreatment(value.toString());
+        setFormData((prev) => ({
+            ...prev,
+            treatment: value.toString(),
+        }));
+        onTreatmentSelect(value);
+    };
+
+    // when dates or treatment change, launch the request
+    useEffect(() => {
+        if (formData.startDate && formData.endDate) {
+            onDateChange?.({
```

Comment:
💭  why this methods is optional? 🤔 shouldn't be mandatory in order to run this component?